nexus/db-queries/src/db/datastore/vpc.rs

```diff
@@ -940,11 +947,26 @@ impl DataStore {
                 "deletion failed due to concurrent modification",
             ));
         } else {
-            self.vpc_system_router_ensure_subnet_routes(
-                opctx,
-                db_subnet.vpc_id,
-            )
-            .await?;
+            // Now, clean up the system route.
+            use db::schema::router_route::dsl as rr_dsl;
+            diesel::update(rr_dsl::router_route)
```

Comment:
Similar to my comment above - the system route will exist briefly after the subnet is deleted, and the RPW version will get updated non-atomically afterwards. Is that okay?

(also, the RPW version might *not* get updated, and the system route might *not* get cleaned up if we crash - but I'm assuming that's part of your comments about moving this into a saga)

Reply:
> the system route will exist briefly after the subnet is deleted, and the RPW version will get updated non-atomically afterwards. Is that okay?

That's okay in this case -- if the RPW comes around early, the subnet lookup will fail and no route will be present in the reified set sent to sled-agent. It'll end up recomputing routes when the RPW version is bumped, but the output set should be identical.

> (also, the RPW version might not get updated, and the system route might not get cleaned up if we crash - but I'm assuming that's part of your comments about moving this into a saga)

That's considerably less okay, hence the need for a saga in return. My concern here is that any leftover routes end up polluting the namespace in a way that can't be directly fixed by users.